create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinContributorExtractor.java

```diff
@@ -0,0 +1,103 @@
+package no.sikt.nva.scopus.conversion;
+
+import static java.util.Objects.nonNull;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import no.scopus.generated.AuthorTp;
+import no.scopus.generated.PersonalnameType;
+import no.sikt.nva.scopus.conversion.model.cristin.Affiliation;
+import no.sikt.nva.scopus.conversion.model.cristin.Person;
+import no.sikt.nva.scopus.conversion.model.cristin.TypedValue;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.Role;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+public final class CristinContributorExtractor {
+
+    public static final String NAME_DELIMITER = ", ";
+    public static final String FIRST_NAME_CRISTIN_FIELD_NAME = "FirstName";
+    public static final String LAST_NAME_CRISTIN_FIELD_NAME = "LastName";
+    public static final String ORCID_FIELD_NAME = "orcid";
+
+    @JacocoGenerated
+    private CristinContributorExtractor() {
+    }
+
+    public static Contributor generateContributorFromCristin(Person person, AuthorTp authorTp,
+                                                             PersonalnameType correspondencePerson) {
+        return new Contributor(generateContributorIdentityFromCristinPerson(person),
+                               generateOrganizationsFromCristinAffiliations(person.getAffiliations()), Role.CREATOR,
+                               getSequenceNumber(authorTp), isCorrespondingAuthor(authorTp, correspondencePerson));
+    }
+
+    private static Identity generateContributorIdentityFromCristinPerson(Person cristinPerson) {
+        var identity = new Identity();
+        identity.setName(determineContributorName(cristinPerson));
+        identity.setOrcId(cristinPerson
+                              .getIdentifiers()
+                              .stream()
+                              .filter(CristinContributorExtractor::isOrcid)
+                              .findAny().map(TypedValue::getValue)
+                              .orElse(null));
+        identity.setId(cristinPerson.getId());
+        return identity;
+    }
+
+    private static List<Organization> generateOrganizationsFromCristinAffiliations(Set<Affiliation> affiliations) {
+        return affiliations
+                   .stream()
+                   .map(affiliation ->
+                            new Organization
+                                    .Builder()
+                                .withId(affiliation.getOrganization())
+                                .withLabels(affiliation.getRole().getLabels())
```

Comment:
roles is always present?

Reply:
Dette er Cristin-proxyen din vi spør. Men vi tar hintet og slenger på en nullsjekk på getRole.

Reply:
Jeg husker ikke om den er nullable, så kanskje best for å være sikker